modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/scopes/ScopeExecutionTestCase.java

```diff
@@ -56,13 +60,15 @@ public void verifyProcessorInitialise() throws Exception {
 
   @Test
   public void verifySameProcessorInstance() throws Exception {
-    Object getChainFirst = runFlow("getChain").getMessage().getPayload().getValue();
-    Object getChainSecond = runFlow("getChain").getMessage().getPayload().getValue();
+    HasMessageProcessors getChainFirst = (HasMessageProcessors) runFlow("getChain").getMessage().getPayload().getValue();
+    HasMessageProcessors getChainSecond = (HasMessageProcessors) runFlow("getChain").getMessage().getPayload().getValue();
     assertThat(getChainFirst, is(not(sameInstance(getChainSecond))));
 
-    Object firstChain = ClassUtils.getFieldValue(getChainFirst, "chain", false);
-    Object secondChain = ClassUtils.getFieldValue(getChainSecond, "chain", false);
-    assertThat(firstChain, is(sameInstance(secondChain)));
+    assertThat("Multiple executions of the same scope should yield the same processor instances",
+               getChainFirst.getMessageProcessors(),
+               contains(getChainSecond.getMessageProcessors().stream()
+                   .map(IsSame::sameInstance)
+                   .collect(toList())));
```

Comment:
Note that this is not exactly the same assertion as before.
We were comparing the MessageProcessorChain instances inside each SDK Chain.
Now we are just comparing the instances of each Processor inside each MPC.

I suppose this should be enough for our actual functional requirements. If you think it is not the case, then I guess I can add a getter for the MPC and move on. After all, the SDK Chain impls are internal.

Reply:
ok